bundles/CoreBundle/src/DependencyInjection/PimcoreCoreExtension.php

```diff
@@ -413,4 +419,25 @@ private function configureAdapterFactories(ContainerBuilder $container, array $f
 
         $serviceLocator->setArgument(0, $arguments);
     }
+
+    /**
+     * Extract class definitions and prefixes if configuration has more than just a class definition
+     */
+    private function flattenConfigurationForTypeLoader(array $configuration) : array
+    {
+        $newConfiguration = [];
+        foreach ($configuration['map'] as $type => $config) {
+            if (array_key_exists('class', $config)) {
+                $newConfiguration['map'][$type] = $config['class'];
```

Comment:
This could cause a warning right, since `map` was not defined as an array first.

Reply:
I added some checks to see if the key exists and also a predefined key.